lib/segment/src/index/hnsw_index/hnsw.rs

```diff
@@ -939,6 +939,11 @@ impl HNSWIndex {
         params: Option<&SearchParams>,
         vector_query_context: &VectorQueryContext,
     ) -> OperationResult<Vec<ScoredPointOffset>> {
+        if vector_query_context.strict_mode_enabled() && vector.is_multivector() {
+            return Err(OperationError::validation_error(
+                "Multivector full scan search not allowed in strict mode",
+            ));
+        }
```

Comment:
Can we move both of these if-statements into the shared `search_plain_iterator` instead?